src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryContext.java

```diff
@@ -392,6 +392,84 @@ public boolean delete(Object pathObject, StarlarkThread thread)
     }
   }
 
+  @StarlarkMethod(
+      name = "rename",
+      doc =
+          """
+          Renames the file or directory from <code>src</code> to <code>dst</code>. \
+          Parent directories are created as needed. Fails if the destination path
+          already exists. Both paths must be located within the repository.
+          """,
+      useStarlarkThread = true,
+      parameters = {
+        @Param(
+            name = "src",
+            allowedTypes = {
+              @ParamType(type = String.class),
+              @ParamType(type = Label.class),
+              @ParamType(type = StarlarkPath.class)
+            },
+            doc =
+                """
+                The path of the existing file or directory to rename, relative
+                to the repository directory.
+                """),
+        @Param(
+            name = "dst",
+            allowedTypes = {
+              @ParamType(type = String.class),
+              @ParamType(type = Label.class),
+              @ParamType(type = StarlarkPath.class)
+            },
+            doc = """
+            The new name to which the file or directory will be renamed to,
+            relative to the repository directory.
+            """),
+      })
+  public void rename(Object srcName, Object dstName, StarlarkThread thread)
```

Comment:
sorry for the late comment here (I was OOO) -- any reason why this is 1) called `rename` instead of `move` (which is the more common name for this operation)? 2) restricted to paths within the current repo (which no other method is)?

Reply:
+1 on calling this `move` now that you mention it.

Restricting this to the repo directory seems reasonable to me as it's the only method that is "destructive" towards its source, which would otherwise make it prone to non-reproducible side effects. 

For example, moving a system file into the current repo wouldn't work a second time and moving files from other repos into the current one would alter the content of the other repo as a side effect of evaluating the current one. Do you have an example of a use case where this wouldn't be a bad idea?

Reply:
Regarding the name, I followed the example of Python's [`os.rename()`](https://docs.python.org/3/library/os.html#os.rename) and POSIX [`rename()`](https://pubs.opengroup.org/onlinepubs/9799919799/functions/rename.html). I don't mind calling this method `move()` instead of `rename()`, but it might be a bit more difficult for users to find and remember.

Other methods that modify the filesystem (notably `delete()`) are also restricted to paths within the repository, for the reasons fmeum described.

Reply:
Good point, Python's `shutil.move` also supports moving between filesystems. But of course that's not relevant for operations within a single repo dir.

Reply:
Good point, Python's `shutil.move` also supports moving between filesystems. But of course that's not relevant for operations within a single repo dir.

Reply:
ok, I see. Sorry for the delay!